app/src/main/java/com/eveningoutpost/dexdrip/Models/Tomato.java

```diff
@@ -17,33 +18,34 @@
 public class Tomato {
     private static final String TAG = "DexCollectionService";//?????"Tomato";
 
-    private static enum TOMATO_STATES {
+    private static final String CHECKSUM_FAILED = "checksum failed";
+
+    private enum TOMATO_STATES {
         REQUEST_DATA_SENT,
         RECIEVING_DATA
     }
-    final static int TOMATO_HEADER_LENGTH = 18;
+    private final static int TOMATO_HEADER_LENGTH = 18;
 
     
-    static TOMATO_STATES s_state;
-    
-    static long s_lastReceiveTimestamp;
-    private static byte[] s_full_data = null;
-    private static int s_acumulatedSize = 0;
-    private static boolean s_recviedEnoughData;
-    
-    
+    static volatile TOMATO_STATES s_state;
     
```

Comment:
When I wrote this file, I was under the assumption that the class will not be accessed by two threads simultaneously. 
If this is not correct, we need to add locks on the function of this class.
Can we trust the dexcollectionservice to sync this readings?

Reply:
I don't know whether there is a situation where we might receive inbound bluetooth messages on multiple threads simultaneously, but even if they are different threads not simultaneously then I think there are possibilities for a state variable not declared volatile to cache a wrong value. 

I don't think there is a performance issue with the processing of this variable so marking it volatile seemed the safest way to deal with this. Its especially hard to simulate because different android implementations can handle bluetooth in different ways and have their own implementations. For example Samsung is often more picky about threads.